library/lua/dfhack.lua

```diff
@@ -86,17 +99,36 @@ function qerror(msg, level)
     dfhack.error(msg, (level or 1) + 1, false)
 end
 
-function dfhack.with_finalize(...)
-    return dfhack.call_with_finalizer(0,true,...)
+---@generic T
+---@generic S
+---@param cleanup_fn fun(...): S
+---@param fn fun(...): T
+---@param ... any
+---@return T|S
```

Comment:
The return value from `cleanup_fn` is not propagated here, just the return values from `fun`

same with `with_onerror` below

Reply:
I misread the docs!